autobahn/asyncio/component.py

```diff
@@ -312,101 +315,118 @@ def start(self, loop=None):
         # specified in the docstring.
         done_f = txaio.create_future()
 
-        # transports to try again and again ..
-        transport_gen = itertools.cycle(self._transports)
+        connectable_transports = dict()
+
+        # Create a generator of transports that .can_reconnect()
+        def set_connectable_transports():
+            valid_transports = [tp for tp in self._transports if tp.can_reconnect()]
+            connectable_transports['count'] =  len(valid_transports)
+            connectable_transports['gen'] = itertools.cycle(valid_transports)
+
+        set_connectable_transports()
 
```

Comment:
Yes it can! Now I know about it  :smile:    I only saw ```can_reconnect``` not ```_can_reconnect``` and thought I had to keep track of remaining connectable transports in the generator so I was removing any that had reached max_retries, but I'll change to ```_can_reconnect```.